src/type/schema.ts

```diff
@@ -350,6 +357,42 @@ export class GraphQLSchema {
     return this.getDirectives().find((directive) => directive.name === name);
   }
 
+  /**
+   * This method looks up the field on the given type definition.
+   * It has special casing for the three introspection fields, `__schema`,
+   * `__type` and `__typename`.
+   *
+   * `__typename` is special because it can always be queried as a field, even
+   * in situations where no other fields are allowed, like on a Union.
+   *
+   * `__schema` and `__type` could get automatically added to the query type,
+   * but that would require mutating type definitions, which would cause issues.
+   */
+  getField(
+    parentType: GraphQLCompositeType,
+    fieldName: string,
+  ): GraphQLField<unknown, unknown> | undefined {
+    switch (fieldName) {
+      case SchemaMetaFieldDef.name:
+        return this.getQueryType() === parentType
+          ? SchemaMetaFieldDef
+          : undefined;
+      case TypeMetaFieldDef.name:
+        return this.getQueryType() === parentType
+          ? TypeMetaFieldDef
+          : undefined;
+      case TypeNameMetaFieldDef.name:
+        return TypeNameMetaFieldDef;
+    }
+
+    // this function is part "hot" path inside executor and check presence
+    // of 'getFields' is faster than to use `!isUnionType`
```

Comment:
how much faster? just in general? is there a reference as to why? just curious....

Reply:
@yaacovCR 
<img width="618" alt="image" src="https://user-images.githubusercontent.com/8336157/170344002-c7af9dc3-da17-43a9-aa07-c5b0ea2596a4.png">

Reply:
very interesting, I tried it with naked `instanceof` and performance regression is still there just significantly smaller:
<img width="747" alt="image" src="https://user-images.githubusercontent.com/8336157/170351323-a618ac30-2b8f-4d51-8e79-463ebcb6b941.png">

Reply:
What about memoizied instanceOf or instanceof? Just curious about all this for discussion about replacing instanceof

Reply:
<img width="348" alt="image" src="https://user-images.githubusercontent.com/8336157/170776465-aa0df036-b1cf-42dd-bb46-48f127a0c8a9.png">